ledger/catchpointtracker.go

```diff
@@ -555,16 +555,19 @@ func (ct *catchpointTracker) commitRound(ctx context.Context, tx trackerdb.Trans
 		}
 
 		var trie *merkletrie.Trie
+		ct.catchpointsMu.Lock()
 		if ct.balancesTrie == nil {
 			trie, err = merkletrie.MakeTrie(mc, trackerdb.TrieMemoryConfig)
 			if err != nil {
 				ct.log.Warnf("unable to create merkle trie during committedUpTo: %v", err)
+				ct.catchpointsMu.Unlock()
 				return err
 			}
 			ct.balancesTrie = trie
 		} else {
 			ct.balancesTrie.SetCommitter(mc)
 		}
+		ct.catchpointsMu.Unlock()
```

Comment:
Since this will throw a runtime error if invoked and already unlocked, is it more obvious to the reader to put this in the else and outside of `err != nil` block?

Reply:
Probably not, as we hit this same case a couple hundred lines down.

Reply:
it is possible to add something like
```golang
defer func() {
  if needUnlock {
    ct.catchpointsMu.Unlock()
  }
```
and `needUnlock` in couple places.